File: packages/kotlin/src/android/src/main/java/alphaTab/importer/alphaTex/SimpleAlphaTexParameterDefinition.kt
package alphaTab.importer.alphaTex

internal class SimpleAlphaTexParameterDefinition {
    val length: Double
    val v0: alphaTab.collections.List<AlphaTexNodeType>
    val v1: ArgumentListParseTypesMode
    val v2: alphaTab.collections.List<String>?
    val v3: alphaTab.collections.List<String>?

    public constructor(
        types: alphaTab.collections.List<AlphaTexNodeType>,
        parseMode: ArgumentListParseTypesMode
    ) {
        v0 = types
        v1 = parseMode
        v2 = null
        v3 = null
        length = 2.0
    }

    public constructor(
        types: alphaTab.collections.List<AlphaTexNodeType>,
        parseMode: ArgumentListParseTypesMode,
        allowedValues: alphaTab.collections.List<String>?
    ) {
        v0 = types
        v1 = parseMode
        v2 = allowedValues
        v3 = null
        length = 3.0
    }

    public constructor(
        types: alphaTab.collections.List<AlphaTexNodeType>,
        parseMode: ArgumentListParseTypesMode,
        allowedValues: alphaTab.collections.List<String>?,
        reservedIdentifiers: alphaTab.collections.List<String>?
    ) {
        v0 = types
        v1 = parseMode
        v2 = allowedValues
        v3 = reservedIdentifiers
        length = 4.0
    }
}

File: packages/csharp/src/AlphaTab/Importer/AlphaTex/SimpleAlphaTexParameterDefinition.cs
using System.Collections.Generic;

namespace AlphaTab.Importer.AlphaTex;

internal class SimpleAlphaTexParameterDefinition
{
    public double Length { get; }
    public IList<AlphaTexNodeType> V0 { get; }
    public ArgumentListParseTypesMode V1 { get; }
    public IList<string>? V2 { get; }
    public IList<string>? V3 { get; }

    public SimpleAlphaTexParameterDefinition(
        IList<AlphaTexNodeType> types,
        ArgumentListParseTypesMode parseMode
        )
    {
        V0 = types;
        V1 = parseMode;
        V2 = null;
        V3 = null;
        Length = 2;
    }

    public SimpleAlphaTexParameterDefinition(
        IList<AlphaTexNodeType> types,
        ArgumentListParseTypesMode parseMode,
        IList<string>? allowedValues)
    {
        V0 = types;
        V1 = parseMode;
        V2 = allowedValues;
        V3 = null;
        Length = 3;
    }

    public SimpleAlphaTexParameterDefinition(
        IList<AlphaTexNodeType> types,
        ArgumentListParseTypesMode parseMode,
        IList<string>? allowedValues,
        IList<string>? reservedIdentifiers)
    {
        V0 = types;
        V1 = parseMode;
        V2 = allowedValues;
        V3 = reservedIdentifiers;
        Length = 4;
    }
}
